### PR TITLE
gdu: enable tests and fix version output

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Enable tests again.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113304` at 3d659902 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/e63aef7f0163bba10698189e1a497747">gdu</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md">no-flags-array</a>

`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/tools/system/gdu/default.nix:20:3:
```
   |
20 |   buildFlagsArray = [ "-ldflags=-s -w -X github.com/dundee/gdu/build.Version=${version}" ];
   |   ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Can you mark the package broken on aarch64?

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113304` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gdu</li>
  </ul>
</details>

**Comment by fabaff** [comment]
> Can you mark the package broken on aarch64? Or not. Not sure why ofborg failed and the other bot not.

There ware a issue with the tests on Darwin. Could be the same now with aarch64.

**Comment by zowoq** [comment]
> There ware a issue with the tests on Darwin. 

Tests are still broken on darwin:
```
# github.com/dundee/gdu/cmd [github.com/dundee/gdu/cmd.test]
cmd/run_test.go:101:12: undefined: device.LinuxDevicesInfoGetter
cmd/run_test.go:136:12: undefined: device.LinuxDevicesInfoGetter
FAIL    github.com/dundee/gdu/cmd [build failed]
```

**Comment by SuperSandro2000** [comment]
Do we want to mark it broken for darwin or merge it without that?

File: pkgs/tools/system/gdu/default.nix [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+    substituteInPlace cmd/run_test.go --replace "development" "${version}"
+  '';
+
   postInstall = ''
     installManPage gdu.1
   '';
 
-  # tests fail if the version is set
-  doCheck = false;
+  doCheck = !stdenv.isAarch64;
```
Comment (zowoq) [inline pkgs/tools/system/gdu/default.nix:38]: We can keep the test disabled for darwin.

```suggestion
  doCheck = !(stdenv.isAarch64 || stdenv.isDarwin);
```

Comment (fabaff) [reply pkgs/tools/system/gdu/default.nix]: Done, thanks.

**Review by tomberek — DISMISSED** [review]
Needs an update for vendorSha256 and rebase

**Comment by fabaff** [comment]
This PR fixes also the broken version output which was detected in https://github.com/NixOS/nixpkgs/pull/114654.

File: pkgs/tools/system/gdu/default.nix [hunk 2]
```diff
@@ -15,18 +16,26 @@ buildGoModule rec {
...
-  buildFlagsArray = [ "-ldflags=-s -w -X github.com/dundee/gdu/build.Version=${version}" ];
+  vendorSha256 = "1aw9cgm9m3bflncpfk2qsid2aqs710nn2bpxx46n54kw9jkvj8s2";
 
   nativeBuildInputs = [ installShellFiles ];
 
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X github.com/dundee/gdu/v${lib.strings.substring 0 1 version}/build.Version=${version}"
```
Comment (zowoq) [inline pkgs/tools/system/gdu/default.nix:27]:
```suggestion
    "-X github.com/dundee/gdu/v${lib.versions.major version}/build.Version=${version}"
```

Comment (fabaff) [reply pkgs/tools/system/gdu/default.nix]: Thanks

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113304` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gdu</li>
</ul>
</details>

**Comment by fabaff** [comment]
Thanks, SuperSandro2000 for checking on darwin.

**Pull request merged**